torchdynamo/optimizations/python_key.py

```diff
@@ -121,13 +139,32 @@ def unpack(x):
 
         return tuple(unpack(x) for x in out)
 
+    class DynamoPythonKeyTracer(PythonKeyTracer):
+        def __init__(self):
+            super().__init__()
+
+        def _module_getattr(self, attr, attr_val, parameter_proxy_cache):
```

Comment:
@jansel I was incorrect earlier - we don't need to copy, but we do need to provide our own `_module_getattr` over the `Tracer` one. We could implement it like this, OR change the main torch.fx `Tracer` to accept a `python_tensor_cls`. Do you have a preference?

Reply:
I was going to say we change behavior too much to call super, then I realized that kind of violates Liskov, so let me actually add this to the base Tracer, it feels more correct as such.